src/hyperion/device_setup_plans/setup_panda.py

```diff
@@ -0,0 +1,153 @@
+) -> SeqTable:
+    """
+
+    -Setting a 'signal' means trigger PCAP internally and send signal to Eiger via physical panda output
+    -When we wait for the position to be greater/lower, give some safe distance (X_STEP_SIZE/2 * MM_TO_ENCODER counts) as the encoder counts arent always exact
+    SEQUENCER TABLE:
+        1:Wait for physical trigger from motion script to mark start of scan / change of direction
+        2:Wait for POSA (X2) to be greater than X_START, then
+            send a signal out every (minimum eiger exposure time + eiger dead time)
+        3:Wait for POSA (X2) to be greater than X_START + X_STEP_SIZE + a bit of leeway for the final trigger, then cut out the signal
+        4:Wait for physical trigger from motion script to mark change of direction
+        5:Wait for POSA (X2) to be less than X_START + X_STEP_SIZE + EXPOSURE_DISTANCE, then
+            send a signal out every (minimum eiger exposure time + eiger dead time)
+        6:Wait for POSA (X2) to be less than (X_START - some leeway + EXPOSURE_DISTANCE), then cut out signal
+        7:Go back to step one.
+
+        For a more detailed explanation and a diagram, see https://github.com/DiamondLightSource/hyperion/wiki/PandA-constant%E2%80%90motion-scanning
+    """
+
+    sample_velocity_mm_per_s = parameters.x_step_size * 1e-3 / time_between_x_steps_ms
+
+    row1 = SeqTableRow(trigger=SeqTrigger.BITA_1, time2=1)
+    row2 = SeqTableRow(
+        trigger=SeqTrigger.POSA_GT,
+        position=int(parameters.x_start * MM_TO_ENCODER_COUNTS),
+        time2=1,
+        outa1=True,
+        outa2=True,
+    )
+    row3 = SeqTableRow(
+        position=int(
+            (parameters.x_start * MM_TO_ENCODER_COUNTS)
+            + (
+                parameters.x_step_size
+                * (
+                    parameters.x_steps - 1
+                )  # x_start is the first trigger point, so we need to travel to x_steps-1 for the final triger point
+                * MM_TO_ENCODER_COUNTS
+                + (MM_TO_ENCODER_COUNTS * (parameters.x_step_size / 2))
+            )
+        ),
+        trigger=SeqTrigger.POSA_GT,
+        time2=1,
+    )
+
+    row4 = SeqTableRow(trigger=SeqTrigger.BITA_1, time2=1)
+
+    row5 = SeqTableRow(
+        trigger=SeqTrigger.POSA_LT,
+        position=(parameters.x_start * MM_TO_ENCODER_COUNTS)
+        + (
+            parameters.x_step_size * (parameters.x_steps - 1) * MM_TO_ENCODER_COUNTS
+            + (sample_velocity_mm_per_s * exposure_time_s * MM_TO_ENCODER_COUNTS)
+        ),
+        time2=1,
+        outa1=True,
+        outa2=True,
+    )
+
+    row6 = SeqTableRow(
+        trigger=SeqTrigger.POSA_LT,
+        position=parameters.x_start * MM_TO_ENCODER_COUNTS
+        - (MM_TO_ENCODER_COUNTS * (parameters.x_step_size / 2))
+        + (sample_velocity_mm_per_s * exposure_time_s * MM_TO_ENCODER_COUNTS),
+        time2=1,
+    )
+
+    rows = [row1, row2, row3, row4, row5, row6]
```

Comment:
Nit: I think this is more readable than before but the use of `rowX` throughout isn't that great. Could you either name each row something related to its function or just do successive `row.append` calls? you could probably also improve readability by making some intermediate variables e.g. `x_start_counts`, `x_end_counts`, `safe_distance`

Reply:
Good idea, it's much more readable with the intermediate variables